aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -42,7 +45,7 @@ struct Reduction {
           index_t result_index = 0;
           for (int64_t k = 0; k < n; k++) {
             scalar_t value = data[k];
-            bool cmp = greater ? (result > value) : (result < value);
+            bool cmp = greater ? (zabs_(result) > zabs_(value)) : (zabs_(result) < zabs_(value));
```

Comment:
abs-based max/min; makes sense, I suppose

Reply:
is that what numpy does?  Why is there no comparison to what NumPy does in any of these?

I don't think it is, e.g:
```
>>> np.max(np.array([0, -5, 2, 3]).astype(np.complex64))
\(3+0j)
```

Reply:
Hi @gchanan,

The test cases are found here: [pytorch-cpu-strided-complex extension](https://gitlab.com/pytorch-complex/pytorch-cpu-strided-complex).

The 'test/test_torch.py' compares the pytorch to numpy for each math kernel

```
import numpy as np
import torch as th

device = th.device('cpu')

def t2n(t):
    return t.detach().numpy()

def test_tensor_compare(self):
     a = th.tensor([[1+1j, 1-1j], [0.51, 0.51+0.51j]], dtype=th.complex128, device=device)
     assert_allclose(th.max(a, -2)[0], np.max(t2n(a), axis=-2))
```

This test shows that `th.max()` is the same as `np.max()`.

I can't generate random numbers with `torch.rand()` because it's still implemented in the legacy TH folder. If that kernel was ported over I could  simply modify the internal test scripts.

FYI, here is a list of legacy kernels that are causing me the most pain:

```
    print(a)  # _th_masked_select_bool (you can get print to work by removing formatting code)
    th.zeros_like(a) # th_zero
    th.randint_like(c8)  # _th_random_
    th.rand((4, 4), dtype=th.complex128, device=device)  # th_uniform
    th.randn((4, 4), dtype=th.complex128, device=device)  # th_normal

    th.mv(m2, v2)  # _th_mv not implemented
    th.mm(m2, m2)  # _th_mm not implemented
```
I'm not sure any of the indexing functions work. Eventually, I can work on porting kernels over from TH, but I'm currently under pressure to demonstrate CPU, GPU, FPGA interoperability.